lib/monkey/deps/flb_libco/ppc64le.c

```diff
@@ -0,0 +1,253 @@
+  0x7f042bce, /* stvxl v24, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7f242bce, /* stvxl v25, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7f442bce, /* stvxl v26, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7f642bce, /* stvxl v27, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7f842bce, /* stvxl v28, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7fa42bce, /* stvxl v29, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7fc42bce, /* stvxl v30, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7fe42bce, /* stvxl v31, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7ca042a6, /* mfvrsave r5       */
+  0x90a402e0, /* stw r5, 736(4)    */
+  0x38a00210, /* li r5, 528      */
+  0x7e832ace, /* lvxl v20, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7ea32ace, /* lvxl v21, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7ec32ace, /* lvxl v22, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7ee32ace, /* lvxl v23, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7f032ace, /* lvxl v24, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7f232ace, /* lvxl v25, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7f432ace, /* lvxl v26, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7f632ace, /* lvxl v27, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7f832ace, /* lvxl v28, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7fa32ace, /* lvxl v29, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7fc32ace, /* lvxl v30, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7fe32ace, /* lvxl v31, r3, 5  */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x80a302d0, /* lwz r5, 720(3)  */
+  0x7ca043a6, /* mtvrsave r5     */
+  #endif
+
+  0x4e800020  /* blr             */
+};
+
+/* function call goes directly to code */
+#define CO_SWAP_ASM(x, y) ((void (*)(cothread_t, cothread_t))(uintptr_t)libco_ppc_code)(x, y)
+
+static uint64_t* co_create_(unsigned size, uintptr_t entry) {
+  (void)entry;
+
+  uint64_t* t = (uint64_t*)malloc(size);
+
+  return t;
+}
+
+cothread_t co_create(unsigned int size, void (*entry_)(void),
+                     size_t *out_size) {
+
+  uintptr_t entry = (uintptr_t)entry_;
+  uint64_t* t = 0;
+
+  /* be sure main thread was successfully allocated */
+  if(co_active()) {
+    size += state_size + above_stack + stack_align;
+    t = co_create_(size, entry);
+  }
+
+  if(t) {
+    uintptr_t sp;
+    int shift;
+
+    /* save current registers into new thread, so that any special ones will have proper values when thread is begun */
+    CO_SWAP_ASM(t, t);
+
+    /* put stack near end of block, and align */
+    sp = (uintptr_t)t + size - above_stack;
+    sp -= sp % stack_align;
+
+    /* set up sp and entry will be called on next swap */
+    t[1] = (uint64_t)sp;
+    t[12] = (uint64_t)entry;
+    t[32] = (uint64_t)entry;
+  }
+  *out_size = size;
+  return t;
+}
+
+void co_delete(cothread_t t) {
+  free(t);
+}
+
+static void co_init_(void) {
```

Comment:
Could you please match the `amd64` implementation when it comes to the handling of `co_active_handle`? 
https://github.com/fluent/fluent-bit/blob/0cd42ae735c14597dea441ea06bba61c060fa4b7/lib/monkey/deps/flb_libco/amd64.c#L137

I think there is no reason to deviate from that because it means adding unnecessary complexity.